test/framework/src/main/java/org/elasticsearch/test/rest/yaml/section/ClientYamlTestSuite.java

```diff
@@ -139,39 +142,45 @@ public TeardownSection getTeardownSection() {
+            errors = Stream.concat(errors,
+                validateExecutableSections(testSection.getExecutableSections(), testSection, setupSection, teardownSection));
+        }
+        String errorMessage = errors.collect(Collectors.joining(",\n"));
+        if (errorMessage.isEmpty() == false) {
+            throw new IllegalArgumentException(getPath() + ":\n" + errorMessage);
         }
     }
 
-    private static void validateExecutableSections(ClientYamlTestSuite yamlTestSuite, List<ExecutableSection> sections,
+    private static Stream<String> validateExecutableSections(List<ExecutableSection> sections,
                                                    ClientYamlTestSection testSection,
                                                    SetupSection setupSection, TeardownSection teardownSection) {
-        for (ExecutableSection section : sections) {
-            if (section instanceof DoSection) {
-                DoSection doSection = (DoSection) section;
-                if (false == doSection.getExpectedWarningHeaders().isEmpty()
-                    && false == hasSkipFeature("warnings", testSection, setupSection, teardownSection)) {
-                    throw new IllegalArgumentException(yamlTestSuite.getPath() + ": attempted to add a [do] with a [warnings] section " +
-                        "without a corresponding [\"skip\": \"features\": \"warnings\"] so runners that do not support the [warnings] " +
-                        "section can skip the test at line [" + doSection.getLocation().lineNumber + "]");
-                }
-                if (NodeSelector.ANY != doSection.getApiCallSection().getNodeSelector()
-                    && false == hasSkipFeature("node_selector", testSection, setupSection, teardownSection)) {
-                    throw new IllegalArgumentException(yamlTestSuite.getPath() + ": attempted to add a [do] with a [node_selector] " +
-                        "section without a corresponding [\"skip\": \"features\": \"node_selector\"] so runners that do not support the " +
-                        "[node_selector] section can skip the test at line [" + doSection.getLocation().lineNumber + "]");
-                }
-            }
-            if (section instanceof ContainsAssertion
-                && false == hasSkipFeature("contains", testSection, setupSection, teardownSection)) {
-                throw new IllegalArgumentException(yamlTestSuite.getPath() + ": attempted to add a [contains] assertion " +
-                    "without a corresponding [\"skip\": \"features\": \"contains\"] so runners that do not support the " +
-                    "[contains] assertion can skip the test at line [" + section.getLocation().lineNumber + "]");
-            }
-        }
+
+        Stream<String> errors = sections.stream().filter(section -> section instanceof DoSection)
+            .map(section -> (DoSection) section)
+            .filter(section -> false == section.getExpectedWarningHeaders().isEmpty()
```

Comment:
I find it more readable to  call `.filter()` twice instead of using `&&` , but might be just me.

Reply:
ok I was debating the same, I don't have a preference myself so I will make this change